datafusion/expr/src/logical_plan/ddl.rs

```diff
@@ -303,12 +531,154 @@ pub struct CreateMemoryTable {
     pub or_replace: bool,
     /// Default values for columns
     pub column_defaults: Vec<(String, Expr)>,
-    /// Wheter the table is `TableType::Temporary`
+    /// Whether the table is `TableType::Temporary`
     pub temporary: bool,
 }
 
+impl CreateMemoryTable {
+    pub fn new(fields: CreateMemoryTableFields) -> Result<Self> {
```

Comment:
It would be nice to call this `try_new()` to signal it can return an error

Reply:
done!